Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -259,7 +259,7 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
     if (!hasPointer) {
         // The hover check can be expensive (it may end up doing selector matching), so we only run it on some elements.
         bool hasVisibleBoxDecorations = renderer.hasVisibleBoxDecorations();
-        bool nonScrollable = !renderer.hasPotentiallyScrollableOverflow();
+        bool nonScrollable = !renderer.hasPotentiallyScrollableOverflow() || renderer.style().overflowX() == Overflow::Hidden;
```

Comment:
Only X? What makes X more interesting than Y?

Reply:
`RenderObject::hasPotentiallyScrollableOverflow()` only tests for X, so I wanted to be consistent here.

Reply:
(Not just for consistency, it's actually not needed since this extra check just makes up for the fact that `RenderObject::hasPotentiallyScrollableOverflow()` checks for `overflowX` values)